Model/AHK.cs

```diff
@@ -10,92 +10,159 @@
 
 namespace _4RTools.Model
 {
-
     public class KeyConfig {
-
-        public Key key { get; set; }
-        public bool clickActive { get; set; }
+        public Key Key { get; set; }
+        public bool ClickActive { get; set; }
 
         public KeyConfig(Key key, bool clickAtive) {
-            this.key = key;
-            this.clickActive = clickAtive;
+            this.Key = key;
+            this.ClickActive = clickAtive;
         }
-
     }
 
     public class AHK : Action
     {
-        public Dictionary<string,KeyConfig> ahkEntries { get; set; } = new Dictionary<string, KeyConfig>();
-        private string ACTION_NAME = "AHK20";
-        public int ahkDelay { get; set; } = 10;
-
+        private const string ACTION_NAME = "AHK20";
+        private const string COMPATIBILITY = "ahkCompatibility";
+        private const string COMPATIBILITY_WITHOUT_FLICK = "ahkCompatibilityWithoutFlick";
+        private const string SPEED_BOOST = "ahkSpeedBoost";
+        public Dictionary<string, KeyConfig> AhkEntries { get; set; } = new Dictionary<string, KeyConfig>();
+        public int AhkDelay { get; set; } = 10;
+        public string AhkMode { get; set; } = COMPATIBILITY;
         private _4RThread thread;
 
-
         public AHK()
         {
         }
 
         public void Start()
         {
             Client roClient = ClientSingleton.GetClient();
-            if(roClient != null)
+            if (roClient != null)
             {
-                if (this.ahkDelay <= 0) this.ahkDelay = 1; //TODO -> This is really necessary? NumberInput already do this for us.
-                this.thread = new _4RThread(_ => AHKThreadExecution(roClient));
+                if (thread != null) {
+                    _4RThread.Stop(this.thread);
+                }
```

Comment:
This line fixes a bug when changing profile in some situation, the spammer still running in another thread with the old profile configuration. Now, on every start call, we'll stop the old thread and start a new one with the current profile configuration + AHK mode.